galaxy_ng/app/api/v3/viewsets/collection.py

```diff
@@ -229,6 +230,15 @@ class CollectionArtifactDownloadView(api_base.APIView):
     permission_classes = [access_policy.CollectionAccessPolicy]
     action = 'retrieve'
 
+    def _get_tcp_response(self, url):
+        return requests.get(url, stream=True, allow_redirects=False)
+
+    def _get_unix_socket_response(self, url):
+        socket_file = settings.CONTENT_BIND.split(':')[1]
+        session = requests.Session()
+        session.mount("http://", SocketHTTPAdapter(socket_file))
```

Comment:
Could this session.mount() mount a 'unix://' adapter?

It's been a while since I used the requests / urllib3 Adapter stuff, but I was thinking you
could do that and then just pass in the 'unix://...'  "url" to request.get()